src/domain/ledgerAccountTypes/index.js

```diff
@@ -0,0 +1,100 @@
+/*****
+ License
+ --------------
+ Copyright © 2017 Bill & Melinda Gates Foundation
+ The Mojaloop files are made available by the Bill & Melinda Gates Foundation under the Apache License, Version 2.0 (the "License") and you may not use these files except in compliance with the License. You may obtain a copy of the License at
+ http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, the Mojaloop files are distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ Contributors
+ --------------
+ This is the official list of the Mojaloop project contributors for this file.
+ Names of the original copyright holders (individuals or organizations)
+ should be listed with a '*' in the first column. People who have
+ contributed from an organization can be listed under the organization
+ that actually holds the copyright for their contributions (see the
+ Gates Foundation organization for an example). Those individuals should have
+ their names indented and be marked with a '-'. Email address can be added
+ optionally within square brackets <email>.
+ * Gates Foundation
+ - Name Surname <name.surname@gatesfoundation.com>
+
+ * ModusBox
+ - Claudio Viola <claudio.viola@modusbox.com>
+ --------------
+ ******/
+'use strict'
+
+const LedgerAccountTypeModel = require('../../models/ledgerAccountType/ledgerAccountType')
+const ErrorHandler = require('@mojaloop/central-services-error-handling')
+const ParticipantFacade = require('../../models/participant/facade')
+const ParticipantPosition = require('../../models/participant/participantPosition')
+const ParticipantCurrency = require('../../models/participant/participantCurrency')
+// const ParticipantCurrencyCached = require('../../models/participant/participantCurrencyCached')
+
+// const Db = require('../../lib/db')
+
+// async function create (name, description, isActive = false, isSettleable = false) {
+//   try {
+//     const knex = Db.getKnex()
+//     await knex.transaction(async trx => {
+//       try {
+//         const ledgerAccountTypeId = await LedgerAccountTypeModel.create(name, description, isActive, isSettleable, trx)
+//         if (isSettleable === true) {
+//           await createAssociatedParticipantAccounts(ledgerAccountTypeId, 'ledgerAccountType', trx)
+//           await ParticipantCurrencyCached.invalidateParticipantCurrencyCache()
+//         }
+//         await trx.commit()
+//       } catch (err) {
+//         await trx.rollback()
+//         throw ErrorHandler.Factory.reformatFSPIOPError(err)
+//       }
+//     })
+
+//     return true
+//   } catch (err) {
+//     throw ErrorHandler.Factory.reformatFSPIOPError(err)
+//   }
+// }
+
+async function createAssociatedParticipantAccounts (ledgerAccountTypeId, createdBy, trx = null) {
+  try {
```

Comment:
Since this invoked when a new settlementModel is created so there is no necessity to add the check because these accounts will not exist and if the account exists the database operation will fail